test/open-uri/test_open-uri.rb

```diff
@@ -115,6 +116,13 @@ def test_open_too_many_arg
     assert_raise(ArgumentError) { open("http://192.0.2.1/tma", "r", 0666, :extra) {} }
   end
 
+  def test_tempfile_rest_parameter
+    temp_file = Tempfile.new
+    assert_nothing_raised { open(temp_file, 'a') }
```

Comment:
Please ignore my comment. I just confirmed `open Tempfile.new, 'a'` works fine without `open-uri`. If that's the case, this is totally a bug.